guides/updating/versions/0.4-to-0.5.md

```diff
@@ -0,0 +1,381 @@
+# Migrating from Credo 0.4.x to 0.5.x
+
+This document describes everything you need to know for updating Credo 0.4.x to 0.5.x. If you're not aware of how updating in Credo works make sure to first read the guide on [Updating Credo](/guides/updating/index.md), and the [Update Assistant](/guides/updating/update-assistant.md). Starting from Credo 0.5, we now support migration of tenant storage in addition to the root wallet. If you're using the tenants module, make sure to read the guides carefully.
+
+See [Pull Request 1747](https://github.com/openwallet-foundation/credo-ts/pull/1747) for more information.).
+
+First of all, update your dependencies to the 0.5.x versions. This will also update the needed peer dependencies. **Extension packages are not updated with this command**. You need to update these manually, and make sure they're up to date with the latest version of Credo.
+
+Credo 0.5.0 is a major release, but there haven't been a lot of breaking changes to the stable API, most of the changes are related to OpenID4VC, SD-JWT VC and DIF Presentation Exchange. One notable change is that the Indy SDK has been fully deprecated and removed from Credo. This means that starting from 0.5, you cannot use Credo with the Indy SDK. If you need to migrate your storage from Indy SDK to Aries Askar, see the note below for instructions. Another notable change is the migration of all AnonCreds credential records to be stored as `W3cCredentialRecord` instead of `AnonCredsCredentialRecord`.
+
+Follow the mentioned steps in this document carefully to make the upgrade as smooth as possible.
+
+:::caution
+
+The migration guide only covers how to migrate from 0.4.x to 0.5.x while keeping the same behavior and dependencies. Credo 0.4.0 added support for [Aries Askar](https://github.com/hyperledger/aries-askar), [Indy VDR](https://github.com/hyperledger/indy-vdr) and [AnonCreds RS](https://github.com/hyperledger/anoncreds-rs) as a replacement for the Indy SDK.
+
+Migrating to these new components requires additional migration steps, which need to be closely followed to prevent loss of data. These can be found at the [Update Indy SDK to Askar guide](../update-indy-sdk-to-askar.md).
+
+As noted in the [Update Indy SDK to Askar guide](../update-indy-sdk-to-askar.md), it is very important that the 0.3.x or 0.4x to 0.5.x update is started after migrating from the Indy SDK to Aries Askar.
+
+The rest of this guides will **assume you have already migrated from the Indy SDK to Aries Askar, AnonCreds and Indy VDR**.
+
+:::
+
+:::caution
+
+The following APIs, modules and features are experimental and therefore not covered by the semver versioning in Credo. If you're using any of these features, make sure to use an exact version of Credo (`0.5.0`) instead of a range (`^0.5.0`):
```

Comment:
I feel like we don't really push breaking changes on patch versions for experimental packages (or maybe we do it quite a bit). If that is the case, can we just keep them experimental (i.e. shit may break and use with caution), but just follow semver? Would make usage quite a bit nicer.

Reply:
> I feel like we don't really push breaking changes on patch versions for experimental packages 

I've pushed and released quite some breaking changes related to OpenId4VC in 0.5.x versions. 

> (i.e. shit may break and use with caution), but just follow semver?

I would really like this, but it means that ALL packages would get a major bump, and then it will become quite a mess. So if we want it, we need to start adopting a new version strategy where not all packages follow the same version. But this will get quite messy as well I think.

Or we should just start pushing out WAAY more versions. I've been thinking of adoptiong changesets: https://github.com/changesets/changesets I think this will really help with version management. But  I still think we NEED to keep same version strategy as otherwise I'll become SOO complicated.

Reply:
> I've pushed and released quite some breaking changes related to OpenId4VC in 0.5.x versions.

Yeah that will make it really difficult sadly.

I am not too familiar with changesets, but anything that will make our live easier with items like this is a yes for me!